torch/_dynamo/variables/torch.py

```diff
@@ -900,6 +901,26 @@ def speculate_branch(branch):
             example_value = r.new_empty(
                 [get_fake_value(args[1].as_proxy().node, tx).shape[0], *r.shape]
             )
+        elif self.value.__name__ == "executorch_call_delegate":
```

Comment:
Hi @voznesenskym 
I imagine this won't be the long term approach to handle these special opaque nodes. Meaning, there would need to be a pluggable way for dynamo to take them in without having to hardcode them.

Could you please confirm?

Reply:
Your imagination serves you well, as always. However, we do not have any design yet - I called this out in another chat (you maybe saw) - atm, there are some hierarchical violations. 

They are not special opaque nodes, but they are python operators, registered with the pyop system.  Ops are lower level than dynamo, yet dynamo knows how to trace them. The biggest issue is actually that this is a non core op, and is user provided, yet dynamo knows about it. 

The violation therefore, is not in the dynamo<>pyoperator relationship, which is basically the same as dynamo<>any torch op relationship - but in the relationship of dynamo (and the whole torch stack) with custom ops. And this is not a new discussion, fwiw, this has always been a tension.

Reply:
Sorry I haven't caught up with threads today.
I'm reviewing other of @angelayi 's changes (internal exir/executorch) and was curious whether there was a resolution on this, so I figured ask directly here in the code: I just wanted to get a sense of where your thinking was at the moment in terms of the design, and a bit timeline (I shall look at the other thread!).
Thanks!

Reply:
> this is a non core op, and is user provided

Just would like to clarify, this op is not user provided, and it's an interface we provide for users to interact with backends. What do you consider is core op, other than cond?

Reply:
> a pluggable way for dynamo to take opaque nodes

Is there any ongoing discussion about it? I understand that ideally, we don't want to do too many special cases with the custom op, however, I do think the interface with backends is generic and is a fundamental capability. It can also be applied in eager mode such that we can interact with different backends smoothly.